src/system.rs

```diff
@@ -354,25 +360,95 @@ impl PendingTypes {
     }
 }
 
+/// Represents a hashmap of BuiltIn derived ISL types
+/// for more information: https://amzn.github.io/ion-schema/docs/spec.html#type-system
+static DERIVED_ISL_TYPES: [&str; 8] = [
+    "type::{ name: any, one_of: [ blob, bool, clob, decimal,
+                                    float, int, string, symbol, timestamp,
+                                    list, sexp, struct ] }",
+    "type::{ name: lob, one_of: [ blob, clob ] }",
+    "type::{ name: number, one_of: [ decimal, float, int ] }",
+    "type::{ name: text, one_of: [ string, symbol ] }",
+    "type::{ name: $any, one_of: [ $blob, $bool, $clob, $decimal,
+                                    $float, $int, $string, $symbol, $timestamp,
+                                    $list, $sexp, $struct ] }",
+    "type::{ name: $lob, one_of: [ $blob, $clob ] }",
+    "type::{ name: $number, one_of: [ $decimal, $float, $int ] }",
+    "type::{ name: $text, one_of: [ $string, $symbol ] }",
+];
+
 pub type TypeId = usize;
 
 /// Defines a cache that can be used to store resolved [Type]s of a [Schema]
 #[derive(Debug, Clone)]
 pub struct TypeStore {
-    core_type_ids_by_name: HashMap<String, TypeId>, // stores all the core types used within this schema
+    builtin_type_ids_by_name: HashMap<String, TypeId>, // stores all the builtin types used within this schema
     imported_type_ids_by_name: HashMap<String, TypeId>, // stores all the imported types of a schema
-    ids_by_name: HashMap<String, TypeId>,           // stores named types defined within the schema
+    ids_by_name: HashMap<String, TypeId>, // stores named types defined within the schema
     types_by_id: Vec<TypeDefinition>,
 }
 
 impl TypeStore {
-    pub fn new() -> Self {
-        Self {
-            core_type_ids_by_name: HashMap::new(),
+    pub fn new() -> IonSchemaResult<Self> {
+        let mut type_store = Self {
+            builtin_type_ids_by_name: HashMap::new(),
             imported_type_ids_by_name: HashMap::new(),
             ids_by_name: HashMap::new(),
             types_by_id: Vec::new(),
+        };
+        type_store.preload()?;
```

Comment:
If `preload()` fails, it means there's a bug in the library. The user can't do anything to recover in that case. I think it would be appropriate to panic with a descriptive message instead of returning an `IonSchemaResult`.